server/src/webserver.rs

```diff
@@ -645,15 +655,46 @@ async fn serve_file(
         source_lexer(&file_contents, lexer)
     };
 
-    // <p>Convert doc blocks from Markdown to HTML</p>
+    // <p>Convert doc blocks from Markdown to HTML then parse using HTML5Ever</p>
     let mut options = Options::empty();
     options.insert(Options::ENABLE_STRIKETHROUGH);
     for code_doc_block in &mut code_doc_block_arr {
         if let CodeDocBlock::DocBlock(ref mut doc_block) = code_doc_block {
             let parser = Parser::new_ext(&doc_block.contents, options);
             let mut html_output = String::new();
             html::push_html(&mut html_output, parser);
-            doc_block.contents = html_output;
+
+            // <p>Create a new RcDom object to represent the DOM tree with default settings</p>
+            let dom = RcDom::default();
+            // <p>Convert html_output to UTF-8 encoding then use parse_fragment to parse the HTML with default options:
+            // parse_fragment is a function of HTML5Ever that takes four arguments:
+            // (1) RcDom object representing a DOM tree where the parsed hTML will be added, in this case dom. See html5ever::parse_fragment
+            // (2) ParseOpts object specifies the parsing options, in this case the default options are used. See ParseOpts::default()
+            // (3) QualName represents the name of the root element of the HTML which in this case is <html>.  See QualName::new
+            // (4) SerializerOpts soecufy tge serialization options which we use the default. See Default::default()
+            // Modifers:
+            // (A) from_utf8 converts the parse HTML of parse_fragment from Vector to a String
+            // (B) read_from converts the parsed HTML string into an RcDom object specified by dom
+            // (C) unwrap extracts the ParseResult object representing the parse HTML
+            // </p>
+            let parse_result = html5ever::parse_fragment(
+                dom,
+                ParseOpts::default(),
+                QualName::new(None, ns!(), "html".into()),
+                Default::default(),
+            )
+            .from_utf8()
+            .read_from(&mut html_output.as_bytes())
+            .unwrap();
```

Comment:
For your next PR: provide some way to deal with a failed parse. Report the error to the user? (How? We should discuss). Simply pass the unparsed HTML on? We need some sort of plan/strategy to handle this case.